hardware/plugins/Plugins.cpp

```diff
@@ -626,8 +626,7 @@ namespace Plugins {
 
 	void CPlugin::Notifier(std::string Notifier)
 	{
-		if (m_Notifier)
-			delete m_Notifier;
+		delete m_Notifier;
```

Comment:
Please do not remove the useless checks! Always good to check.
What happens when the system is out of memory, and a new value could not be allocated and Notifier is called again ?
Or what happens when a developer made a mistake and calls notifier twice
It might look unneeded, but it is always good to check for it's state

Reply:
@gizmocuz They are useless not because you can be certain m_Notifier != NULL (you can not as you point out), they are useless because there's no need to pollute the code with NULL-checks before calling `free()` or `delete`.
Here's a SO discussion on the topic with references to the spec: https://stackoverflow.com/a/4190733